app/Services/Nodes/NodeAutoDeployService.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Services\Nodes;
+
+use App\Models\ApiKey;
+use App\Models\Node;
+use App\Services\Api\KeyCreationService;
+use Illuminate\Http\Request;
+
+class NodeAutoDeployService
+{
+    /**
+     * NodeAutoDeployService constructor.
+     */
+    public function __construct(
+        private readonly KeyCreationService $keyCreationService
+    ) {
+    }
+
+    /**
+     * Generates a new API key for the logged-in user with only permission to read
+     * nodes, and returns that as the deployment key for a node.
+     *
+     * @throws \App\Exceptions\Model\DataValidationException
+     */
+    public function handle(Request $request, Node $node, ?bool $docker = false): ?string
+    {
+        /** @var ApiKey|null $key */
+        $key = ApiKey::query()
+            ->where('user_id', $request->user()->id)
```

Comment:
Why the user check? For application keys this shouldn't be needed.

Reply:
It won't find the keys without it thus creating thousand duplicates.

Reply:
Im gonna try again but for some reason it didn't work on my side when i was testing that.